server/embedded/src/org/labkey/embedded/ResourceType.java

```diff
@@ -0,0 +1,113 @@
+                {
+                    String displayName = props.remove("displayName");
+                    if (displayName != null)
+                    {
+                        context.addParameter(name + ":DisplayName", displayName);
+                    }
+                    String logQueries = props.remove("logQueries");
+                    if (logQueries != null)
+                    {
+                        context.addParameter(name + ":LogQueries", displayName);
+                    }
+
+                    super.addResource(name, props, context);
+                }
+            },
+    jms
+            {
+                @Override
+                protected Map<String, String> mergeWithDefaults(Map<String, String> props)
+                {
+                    Map<String, String> result = super.mergeWithDefaults(props);
+                    result.putIfAbsent("type", "org.apache.activemq.ActiveMQConnectionFactory");
+                    result.putIfAbsent("factory", "org.apache.activemq.jndi.JNDIReferenceFactory");
+                    result.putIfAbsent("description", "JMS Connection Factory");
+                    result.putIfAbsent("brokerName", "LocalActiveMQBroker");
+                    return result;
+                }
+            },
+    ldap
+            {
+                @Override
+                protected Map<String, String> mergeWithDefaults(Map<String, String> props)
+                {
+                    Map<String, String> result = super.mergeWithDefaults(props);
+                    result.putIfAbsent("type", "org.labkey.premium.ldap.LdapConnectionConfigFactory");
+                    result.putIfAbsent("factory", "org.labkey.premium.ldap.LdapConnectionConfigFactory");
+                    return result;
+                }
+            },
+    generic;
+
+    protected Map<String, String> mergeWithDefaults(Map<String, String> props)
+    {
+        Map<String, String> result = new CaseInsensitiveKeyMap<>();
+        result.putAll(props);
+        return result;
+    }
+
+    public void addResource(String name, Map<String, String> props, StandardContext context) throws ConfigException
+    {
+        props = mergeWithDefaults(props);
+        ContextResource contextResource = new ContextResource();
+        // Handle resource properties with explicit setters
+        contextResource.setName(name);
+
+        if (!props.containsKey("type"))
+        {
+            throw new ConfigException("Resource configuration error: 'type' is not defined for resource '%s'".formatted(name));
+        }
+
+        contextResource.setType(props.remove("type"));
+        contextResource.setDescription(props.remove("description"));
+        contextResource.setLookupName(props.remove("lookupName"));
+        if (props.containsKey("scope"))
+        {
+            contextResource.setScope(props.remove("scope"));
+        }
+        contextResource.setAuth(Objects.requireNonNullElse(props.remove("auth"), "Container"));
```

Comment:
`mergeWithDefaults` should handle this.
```suggestion
        contextResource.setAuth(props.remove("auth"));
```